File: Python/Algorithms/1893.py
class Solution:
    def isCovered(self, ranges: List[List[int]], left: int, right: int) -> bool:
        # note: could also do with set ops.
        ranges.sort(key=itemgetter(1))
        i, length = 0, len(ranges)
        while left <= right and i < length:
            l, r = ranges[i]
            if l <= left <= r:
                left = r + 1
            if l <= right <= r:
                right = l - 1
            i += 1
        return left > right

File: Javascript/Algorithms/1893.js
/**
 * @param {number[][]} ranges
 * @param {number} left
 * @param {number} right
 * @return {boolean}
 */
var isCovered = function(ranges, left, right) {
  ranges.sort((a, b) => a[1] - b[1]);
  let i = 0;
  while (left <= right && i < ranges.length) {
    let l = ranges[i][0];
    let r = ranges[i][1];
    if (l <= left && left <= r) left = r + 1;
    if (l <= right && right <= r) right = l - 1;
    i++;
  }
  return left > right;
};